tests/pybaseball/datahelpers/caching/test_func_utils.py

```diff
@@ -0,0 +1,122 @@
+
+    def test_get_value_hash_tuple(self) -> None:
+        test_value = ('a', 'b')
+
+        assert caching.get_value_hash(test_value) == "('a', 'b')"
+
+    def test_get_value_hash_tuple_no_designators(self) -> None:
+        test_value = ('a', 'b')
+
+        assert caching.get_value_hash(test_value, include_designators=False) == "'a', 'b'"
+
+    def test_get_value_hash_dict(self) -> None:
+        test_value = {'a': 'foo', 'b': 'bar'}
+
+        assert caching.get_value_hash(test_value) == "{a='foo', b='bar'}"
+
+    def test_get_value_hash_dict_no_designators(self) -> None:
+        test_value = {'a': 'foo', 'b': 'bar'}
+
+        assert caching.get_value_hash(test_value, include_designators=False) == "a='foo', b='bar'"
+
+    def test_get_value_hash_non_hashable(self) -> None:
+        class NonHashable:
+            def __hash__(self) -> int:
+                raise NotImplementedError
+
+        with pytest.raises(ValueError):
+            print(caching.get_value_hash(NonHashable()))
+
+    def test_get_func_name_function(self) -> None:
+        def test_func() -> None:
+            pass
+
+        assert caching.get_func_name(test_func) == "test_func"
+
+    def test_get_func_name_class_method(self) -> None:
+        class test_class:
+            def test_func(self) -> None:
+                pass
+
+        assert caching.get_func_name(test_class().test_func) == "test_class.test_func"
+
+    def test_get_func_hash_no_params(self) -> None:
+        def test_func(*args: Any, **kwargs: Any) -> None:
+            pass
+
+        assert caching.get_func_hash(test_func, (), {}) == "test_func()"
+
+    def test_get_func_hash_with_args(self) -> None:
+        def test_func(*args: Any, **kwargs: Any) -> None:
+            pass
+
+        expected_hash = f"test_func('a', {(1).__hash__()})"
+
+        assert caching.get_func_hash(test_func, ('a', 1), {}) == expected_hash
+
+    def test_get_func_hash_with_kwargs(self) -> None:
+        def test_func(*args: Any, **kwargs: Any) -> None:
+            pass
+
+        expected_hash = f"test_func(val1={(1.5).__hash__()}, val2='b')"
+
+        assert caching.get_func_hash(test_func, (), {'val1': 1.5, 'val2': 'b'}) == expected_hash
+
+    def test_get_func_hash_with_args_and_kwargs(self) -> None:
+        def test_func(*args: Any, **kwargs: Any) -> None:
+            pass
+
+        expected_hash = f"test_func('a', {(1).__hash__()}, val1={(1.5).__hash__()}, val2='b')"
+
+        assert caching.get_func_hash(test_func, ('a', 1), {'val1': 1.5, 'val2': 'b'}) == expected_hash
+
+    def test_get_func_hash_long_path_windows(self) -> None:
+        def test_func(*args: Any, **kwargs: Any) -> None:
+            pass
+
+        expected_hash = "test_func(93ece9cbe326c434412e28b87d466bd08bc0f2e7db0770cff852e7b9f46240d8)"
+
+        long_kwargs = {f"val{str(x)}": x for x in range(caching.MAX_ARGS_KEY_LENGTH)}
+
+        assert caching.get_func_hash(test_func, ('a', 1), long_kwargs) == expected_hash
```

Comment:
Why did we roll our own hash function to begin with, and, given that, why do we have to have 122 lines of unit testing for it?

Reply:
See above.